frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/validations/common-template-validations/common-templates-validations.ts

```diff
@@ -0,0 +1,105 @@
+import { getFieldId } from '../../../utils/vm-settings-tab-utils';
+import { UpdateOptions } from '../../types';
+import { iGetVmSettingAttribute } from '../../../selectors/immutable/vm-settings';
+import { iGetLoadedCommonData } from '../../../selectors/immutable/selectors';
+import { VMSettingsField, VMWizardProps, VMSettingsRenderableField } from '../../../types';
+import {
+  CommonTemplatesValidation,
+  IntegerValidationObj,
+  IntegerValidtionType,
+} from './validation-types';
+import {
+  getRelevantTemplates,
+  getValidationsFromTemplates,
+  getFieldValidations,
+  getValidationAttributes,
+} from './selectors';
+
+// TODO: Add all the fields in the form
+// For each field we need to check for validations
+const IDToJsonPath = {
+  [getFieldId(VMSettingsField.MEMORY)]: '.spec.domain.resources.requests.memory',
+  [getFieldId(VMSettingsField.CPU)]: '.spec.domain.cpu.cores',
+};
+
+export const getTemplateValidations = (
+  options: UpdateOptions,
+  fieldId: VMSettingsRenderableField,
+): CommonTemplatesValidation[][] => {
+  // Proceed if there are no validation for the specific fieldId
+  if (!(fieldId in IDToJsonPath)) {
+    return [];
+  }
+  const { getState, id } = options;
+  const state = getState();
+
+  // Get OS, workload-profile and flavor attributes
+  const os = iGetVmSettingAttribute(state, id, VMSettingsField.OPERATING_SYSTEM);
+  const flavor = iGetVmSettingAttribute(state, id, VMSettingsField.FLAVOR);
+  const workloadProfile = iGetVmSettingAttribute(state, id, VMSettingsField.WORKLOAD_PROFILE);
+
+  // Get all the templates from common-templates
+  const commonTemplates = iGetLoadedCommonData(state, id, VMWizardProps.commonTemplates);
+
+  // Get templates based on OS and WorkloadProfile
+  if (os && workloadProfile) {
```

Comment:
we have this check in `getRelevantTemplates` , it will return an empty array [], and then if `getFieldValidations` return empty array, you will not need this "if" and not the if in line 50